.github/actions/setup-action/action.yml

```diff
@@ -0,0 +1,69 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: 'Setup action'
+description: 'Setup repository and configure the required steps'
+inputs:
+  comment_phrase:
+    description: "The comment phrase in the PR"
+    required: true
+  github_token:
+    description: "The GitHub token"
+    required: true
+  pull_request_url:
+    description: "The URL of the PR"
+    required: false
+    default: ''
+  github_repository:
+    description: "The GitHub repository"
+    required: false
+    default: ''
+  github_job:
+    description: "The GitHub job"
+    required: false
+    default: ''
+  github_current_run_id:
```

Comment:
Question: I know this was partially pulled from the rerun action bit, but can't we just get a lot of these from the github context object? Do we actually need to have them all passed in as params?

Reply:
I removed some of the inputs and left only `github_token` (composite actions don't work with secrets directly) and `github_job` (it is widely used by jobs that have matrix values and the it's generated at runtime). As for the remaining, I think it make sense to leave them as inputs for the rerun action in case we ever need to interact with several repositories. We can revisit the rerun action logic some time later.